src/SourceLocator/Type/AnonymousClassObjectSourceLocator.php

```diff
@@ -136,6 +137,7 @@ public function getAnonymousClassNode() : ?Class_
 
         $nodeTraverser = new NodeTraverser();
         $nodeTraverser->addVisitor($nodeVisitor);
+        $nodeTraverser->addVisitor(new NameResolver());
```

Comment:
Can this change be tested in isolation? The added test case works with `ReflectionClass`